boa/environment.py

```diff
@@ -593,41 +204,30 @@ def deploy_code(
         gas: Optional[int] = None,
         value: int = 0,
         bytecode: bytes = b"",
-        start_pc: int = 0,
+        start_pc: int = 0,  # TODO: This isn't used
         # override the target address:
         override_address: Optional[_AddressType] = None,
     ) -> tuple[Address, bytes]:
-        if gas is None:
-            gas = self.vm.state.gas_limit
-
         sender = self._get_sender(sender)
 
-        if override_address is not None:
-            target_address = Address(override_address)
-        else:
-            nonce = self.vm.state.get_nonce(sender)
-            self.vm.state.increment_nonce(sender)
-            target_address = Address(generate_contract_address(sender, nonce))
+        target_address = (
+            self.evm.generate_contract_address(sender)
+            if override_address is None
+            else Address(override_address)
+        )
```

Comment:
style: please use a branch instead of multi-line ternary

Reply:
I find ternaries easier to read, given it's immediately clear that both `if` and `else` will do an assignment to the same variable. In this case it's clear enough though so np